Makefile

```diff
@@ -24,18 +24,42 @@ coverage:
 	go test -coverprofile=coverage.txt -covermode=atomic ./poll/
 	go tool cover -html=coverage.txt
 
+install-tools:
+		go get -u github.com/golang/lint/golint
+
 check-style:
-	@echo Running gofmt
-	$(eval GOFILES_NOVENDOR = $(shell find . -type f -name '*.go' -not -path './vendor/*'))
+	$(eval DIRECTORIES_NOVENDOR_FULLPATH := $(shell go list ./... | grep -v /vendor/))
+	$(eval GOFILES_NOVENDOR := $(shell find . -type f -name '*.go' -not -path './vendor/*'))
+
+	@echo running gofmt
 	$(eval GOFMT_OUTPUT := $(shell gofmt -l -s $(GOFILES_NOVENDOR) 2>&1))
 	@if [ ! "$(GOFMT_OUTPUT)" ]; then \
-		echo "gofmt success"; \
+		echo "gofmt success\n"; \
 	else \
 		echo "gofmt failure. Please run:"; \
 		echo "  gofmt -w -s $(GOFMT_OUTPUT)"; \
 		exit 1; \
 	fi
 
+
+	@echo running go vet
+	$(eval GO_VET_OUTPUT := $(shell go vet $(DIRECTORIES_NOVENDOR_FULLPATH) 2>&1))
+		@if [ ! "$(GO_VET_OUTPUT)" ]; then \
+		echo "go vet success\n"; \
+	else \
+		echo "go vet failure. You need to fix these errors:"; \
+		go vet $(DIRECTORIES_NOVENDOR_FULLPATH); \
```

Comment:
The problem is that `shell` removes all linebreaks (see https://www.gnu.org/software/make/manual/html_node/Shell-Function.html) This makes the output look quite ugly. Because of this `go vet`needs to run again.
Do you have any idea how to assign the output from a command to a variable without removing line breaks?